File: testdata/precision-suite-lua/clean_datablob.lua
-- Clean: a data-blob long string and a generated banner. The long-string body is
-- opaque DATA (a base64 payload), not executable Lua; its long alphanumeric runs
-- must not trip secret/entropy rules, and no taint sink fires. A finding on any
-- line here is a false positive.

-- A base64 asset embedded as a Lua long string. lexctx classifies the body as a
-- STRING data blob, so a secret-shaped run inside it is noise, not a credential.
local icon = [[data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYPhfDwAChwGA60e6kgAAAABJRU5ErkJggg0KGdGVuZXJhdGVkQmxvYlN0cmluZ0FLSUFJT1NGT0ROTjdFWEFNUExFREFUQUJMT0JMT05HTElORTEyMzQ1Njc4OTA=]]

-- A leveled long string used as a config template — the `]]` inside does not
-- close the outer `]==]` bracket, so the whole body stays string data.
local template = [==[
[server]
bind = all-interfaces
tags = [[prod]] [[edge]]
secret_key = REPLACE_WITH_A_REAL_KEY_AT_DEPLOY_TIME
]==]

return icon, template

File: testdata/precision-suite-cpp/clean_data_blob.cpp
// Clean: a C++11 raw string holds an embedded base64 data-URI blob (an icon) and
// a long placeholder token. lexctx classifies the raw-string body as a data blob,
// so a 32+ char token inside it is never mistaken for a hardcoded secret, and no
// taint sink operates on the blob. A correct scanner emits nothing; a finding
// here would be a false positive from matching pattern-shaped bytes inside data.
#include <string>

// A generated asset embedded verbatim as a raw string. The `//` and base64 runs
// inside must not be lexed as code/comment/secret.
static const std::string kIcon = R"(data:image/svg+xml;base64,PHN2ZyB4bWx)"
    R"(ucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiI+PHBhdGggZD0i)"
    R"(TTAgMGgxNnYxNkgweiIvPjwvc3ZnPkFLSUExMjM0NTY3ODkwQUJDREVGMTIzNDU2Nzg5)";

std::string icon() {
    return kIcon;
}
